eastereggs.cpp

```diff
@@ -99,6 +91,45 @@ namespace RdkShell
         system("systemctl restart wpeframework &");
     }
 
+    bool EasterEgg::invokeEvent()
+    {
+        bool ret = false;
+        if (mSatisfied)
+        {
+            std::vector<std::map<std::string, RdkShell::RdkShellData>> eventData(1);
+            eventData[0] = std::map<std::string, RdkShell::RdkShellData>();
+            eventData[0]["name"] = mName;
+            eventData[0]["action"] = mActionJson;
+            if (mName == "RDKSHELL_FORCE_720")
+            {
+                std::cout << "about to toggle force 720 easter egg\n";
+                toggleForce720();
+            }
+            else
+            {
+                RdkShell::CompositorController::sendEvent("onEasterEgg", eventData);
+            }
+            ret = true;
+        }
+        mSatisfied = false;
+        return ret;
+    }
+
+    void EasterEgg::reset()
+    {
+        mSatisfied = false;
```

Comment:
Should `mCurrentKeyIndex` be set to 0 inside of `reset()` as well to prevent a second easter egg from launching if the full sequence is not entered after another easter egg is triggered?

Reply:
yes, it is needed. Updated . please review the same